### PR TITLE
Rewrite without dolittle

### PR DESCRIPTION
**Pull request by thotokraa**

### CODE REVIEW

File: Source/ByteExtensions.cs [hunk 1]
```diff
@@ -75,7 +66,14 @@ static object ConvertBytes(DataType type, byte[] bytes)
             return 0;
         }
 
-        private static IEnumerable<IEnumerable<T>> Chunk<T>(this IEnumerable<T> source, int size)
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="source"></param>
```
Comment (srudaa) [inline Source/ByteExtensions.cs:73]: What is source?

File: Source/Dockerfile-old [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+COPY Source/ ./
+RUN dotnet publish -c Release -o out
+
+
+FROM base
+
+WORKDIR /app
+COPY --from=build-env /app/Source/out ./
+
+ENTRYPOINT ["dotnet", "RaaLabs.Edge.Connectors.Modbus.dll"]
```
Comment (srudaa) [inline Source/Dockerfile-old:42]: Delete this file?

File: Source/EndiannessExtensions.cs [hunk 3]
```diff
@@ -4,41 +4,70 @@
...
     {
         /// <summary>
-        /// Get correct bytes according to a given <see cref="Endianness"/>
+        /// Gets the bytes of the payload, with corrected endianness.
         /// </summary>
-        /// <param name="shorts">Array of <see cref="ushort">shorts</see></param>
-        /// <param name="endianness"><see cref="Endianness"/> expected in the shorts</param>
-        /// <returns>Array of <see cref="byte"/> with correct endian for current CPU</returns>
-        public static byte[] GetBytes(this ushort[] shorts, Endianness endianness)
+        /// <param name="shorts"></param>
```
Comment (srudaa) [inline Source/EndiannessExtensions.cs:21]: what is shorts?

File: Source/Events/ModbusDatapointOutput.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
+using RaaLabs.Edge.Modules.EdgeHub;
+using System.Diagnostics.CodeAnalysis;
+
+namespace RaaLabs.Edge.Connectors.Modbus.Events
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    [OutputName("output")]
+    [ExcludeFromCodeCoverage]
```
Comment (srudaa) [inline Source/Events/ModbusDatapointOutput.cs:10]: Can remove ExcludeFromCodeCoverage. Tested using HandlerSteps

File: Source/Events/ModbusRegisterReceived.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+using System.Threading.Tasks;
+using RaaLabs.Edge.Connectors.Modbus.Model;
+using System.Diagnostics.CodeAnalysis;
+
+namespace RaaLabs.Edge.Connectors.Modbus.Events
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    [ExcludeFromCodeCoverage]
```
Comment (srudaa) [inline Source/Events/ModbusRegisterReceived.cs:15]: Can remove ExcludeFromCodeCoverage. Tested using HandlerSteps

File: Source/Modbus.csproj [hunk 6]
```diff
@@ -1,21 +1,21 @@
...
 
     <ItemGroup>
         <ProjectCapability Include="AzureIoTEdgeModule" />
     </ItemGroup>
     
-
     <ItemGroup>
-        <PackageReference Include="RaaLabs.TimeSeries.Modules.Connectors" Version="6.*" />
-        <PackageReference Include="RaaLabs.TimeSeries.Modules.IoTEdge" Version="6.*" />
         <PackageReference Include="nmodbus" Version="3.0.58" />
```
Comment (srudaa) [inline Source/Modbus.csproj:15]: Bump to 3.0.62?

File: Source/ModbusRegisterReceivedHandler.cs [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="event"></param>
+        public void Handle(ModbusRegisterReceived @event)
+        {
+            TagWithData[] tagsWithData = @event.Payload.ToTagsWithData(@event.Register);
+
+            var timestamp = (long)(DateTime.UtcNow - new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc)).TotalMilliseconds;
```
Comment (srudaa) [inline Source/ModbusRegisterReceivedHandler.cs:42]: How about this?  DateTimeOffset.UtcNow.ToUnixTimeMilliseconds()

File: Source/ModbusRegisterReceivedHandler.cs [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+                    {
+                        source = "Modbus",
+                        tag = tagWithData.Tag,
+                        timestamp = timestamp,
+                        value = tagWithData.Data
+                    };
+
+                    SendEvent(output);
+
+                    _logger.Information($"Tag: {tagWithData.Tag}, Value : {tagWithData.Data}");
```
Comment (srudaa) [inline Source/ModbusRegisterReceivedHandler.cs:58]: Remove logger. Framework will generate similar log when SendEvent is called.

File: Source/ModbusRegisterReceivedHandler.cs [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Serilog;
+using RaaLabs.Edge.Connectors.Modbus.Model;
+
+namespace RaaLabs.Edge.Connectors.Modbus
+{
+    /// <summary>
+    ///
```
Comment (srudaa) [inline Source/ModbusRegisterReceivedHandler.cs:14]: Add doc

File: Source/Model/TagWithData.cs [hunk 10]
```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace RaaLabs.Edge.Connectors.Modbus.Model
```
Comment (srudaa) [inline Source/Model/TagWithData.cs:8]: Add doc

File: Source/Program.cs [hunk 11]
```diff
@@ -2,15 +2,31 @@
...
+using RaaLabs.Edge.Modules.Configuration;
+using RaaLabs.Edge.Modules.EdgeHub;
+using RaaLabs.Edge.Connectors.Modbus.Bridge;
+using RaaLabs.Edge.Connectors.Modbus;
+using System.Diagnostics.CodeAnalysis;
 
 namespace RaaLabs.TimeSeries.Modbus
 {
+    [ExcludeFromCodeCoverage]
     class Program
```
Comment (srudaa) [inline Source/Program.cs:16]: static class Program

File: Source/data/_streamingConnectors.json [hunk 12]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (srudaa) [inline Source/data/_streamingConnectors.json:1]: Delete?

File: Specs/Features/EndiannessExtensions.feature.cs [hunk 13]
```diff
@@ -0,0 +1,191 @@
+﻿// ------------------------------------------------------------------------------
```
Comment (srudaa) [inline Specs/Features/EndiannessExtensions.feature.cs:1]: Delete file and add *.feature.cs in gitignore

File: Specs/Features/ModbusBridge.feature.cs [hunk 14]
```diff
@@ -0,0 +1,263 @@
...
+//  <auto-generated>
+//      This code was generated by SpecFlow (https://www.specflow.org/).
+//      SpecFlow Version:3.7.0.0
+//      SpecFlow Generator Version:3.7.0.0
+// 
+//      Changes to this file may cause incorrect behavior and will be lost if
+//      the code is regenerated.
+//  </auto-generated>
+// ------------------------------------------------------------------------------
+#region Designer generated code
```
Comment (srudaa) [inline Specs/Features/ModbusBridge.feature.cs:11]: Delete file and add *.feature.cs in gitignore

File: Specs/Features/ModbusRegisterReceivedHandler.feature.cs [hunk 15]
```diff
@@ -0,0 +1,211 @@
+﻿// ------------------------------------------------------------------------------
```
Comment (srudaa) [inline Specs/Features/ModbusRegisterReceivedHandler.feature.cs:1]: Delete file and add *.feature.cs in gitignore

File: Specs/Specs.csproj [hunk 16]
```diff
@@ -0,0 +1,67 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
```
Comment (srudaa) [inline Specs/Specs.csproj:2]: Format doc

**Review by srudaa — CHANGES_REQUESTED** [review]
Great work! 

You can find my comments in each file. 

Summary and other questions: 
1) Please check code smells for Master.cs in sonar cloud. 
2) Add *.feature.cs in gitignore
3) Should it be Specs or Specifications? Other modules like IdentityMapper has Specifications.
4) Code coverage is not reported to sonar cloud.

File: Source/Bridge/Master.cs [hunk 17]
```diff
@@ -6,19 +6,19 @@
...
 
-namespace RaaLabs.TimeSeries.Modbus
+namespace RaaLabs.Edge.Connectors.Modbus.Bridge
 {
     /// <summary>
     /// Represents an implementation of <see cref="IMaster"/>
     /// </summary>
-    [Singleton]
+    //[Singleton]
     public class Master : IMaster, IDisposable
```
[the remark this answers is not in the document]
Comment (thotokraa) [reply Source/Bridge/Master.cs]: I realized that there's no point in making this class implement `IDisposable` anymore, so I removed it.

File: Source/Dockerfile-old (hunk 2 above)
Comment (thotokraa) [reply Source/Dockerfile-old]: Done.

File: Source/EndiannessExtensions.cs (hunk 3 above)
Comment (thotokraa) [reply Source/EndiannessExtensions.cs]: Added comment about this.

File: Source/Events/ModbusDatapointOutput.cs (hunk 4 above)
Comment (thotokraa) [reply Source/Events/ModbusDatapointOutput.cs]: Done.

File: Source/Events/ModbusRegisterReceived.cs (hunk 5 above)
Comment (thotokraa) [reply Source/Events/ModbusRegisterReceived.cs]: Done.

File: Source/Modbus.csproj (hunk 6 above)
Comment (thotokraa) [reply Source/Modbus.csproj]: Done.

File: Source/ModbusRegisterReceivedHandler.cs (hunk 9 above)
Comment (thotokraa) [reply Source/ModbusRegisterReceivedHandler.cs]: Done.

File: Source/Model/TagWithData.cs (hunk 10 above)
Comment (thotokraa) [reply Source/Model/TagWithData.cs]: Removed the class instead, as it was not necessary.

File: Source/Program.cs (hunk 11 above)
Comment (thotokraa) [reply Source/Program.cs]: Done.

File: Source/data/_streamingConnectors.json (hunk 12 above)
Comment (thotokraa) [reply Source/data/_streamingConnectors.json]: Done.

**Review by srudaa — APPROVED** [review]

**Pull request merged**